### PR TITLE
Tie plugin to Spicy version, and analyzers to plugin.

### PR DESCRIPTION
**Pull request by rsmmr**
This uses versions-encoded-in-function names to (1) make sure the plugin gets rebuild if the Spicy version changes, and (2) analyzers get rebuild if the plugin versions changes. Through a separate Spicy-side changes, analyzers are in additional also tied to the Spicy version directly (https://github.com/zeek/spicy/pull/1282)

### CODE REVIEW

**Comment by rsmmr** [comment]
Waiting for https://github.com/zeek/spicy/pull/1282 to land before proceeding.

File: src/plugin.cc [hunk 1]
```diff
@@ -75,6 +77,11 @@ class OurGlueCompiler : public spicy::zeek::GlueCompiler {
 };
 
 plugin::Zeek_Spicy::Plugin::Plugin() {
+    // This ensures version compatibility at dlopen() time  by requiring the
+    // versioned symbol to be present.
+    _spicy_version = HILTI_VERSION_FUNCTION();
```
Comment (bbannier) [inline src/plugin.cc:82]: This seems to be the only aspect requiring >=spicy-1.6.0 and I see you already pulled that into its own commit. Just wondering, do we want to backport the other two changes into the spicy-plugin version we pull into the Zeek LTS release?

File: src/plugin.cc [hunk 2]
```diff
@@ -75,6 +77,11 @@ class OurGlueCompiler : public spicy::zeek::GlueCompiler {
...
+
+    // Not sure if with the with the check above, we still need this? Can't hurt I guess.
```
Comment (bbannier) [inline src/plugin.cc:84]: This seems to check another compatibility aspect, so this should either already have been handled by the Zeek version functionality or we should instead evaluate that symbol like we do with `HILTI_VERSION_FUNCTION`.

**Review by bbannier — DISMISSED** [review]

File: src/plugin.cc (hunk 1 above)
Comment (bbannier) [reply src/plugin.cc:82]: Actually wondering if we shouldn't make this conditional on the Spicy version we are running against (e.g., on the preprocessor level). This should simplify what can go into Zeek. As is we cannot publish any version of the plugin until spicy-1.6.0 is released.

Comment (rsmmr) [reply src/plugin.cc:82]: I'd say let's not backport into the Zeek release branch. But I can put just an `#ifdef` around this, then we can decide what/when to pull upstream.

Comment (bbannier) [reply src/plugin.cc:82]: Sounds good, that should also allow us running CI against more Spicy versions.

File: src/plugin.cc (hunk 2 above)
Comment (rsmmr) [reply src/plugin.cc:84]: Not sure what you mean by "evaluate that symbol like we do with HILTI_VERSION_FUNCTION"?

Comment (bbannier) [reply src/plugin.cc:84]: I am unsure whether the symbol is evaluated automatically with the Zeek-side tooling we are using, or whether we need code like `_spicy_version = HILTI_VERSION_FUNCTION();` but for the Zeek version function.

Comment (rsmmr) [reply src/plugin.cc:84]: Ah, Zeek's plugin integration takes care of calling the Zeek version function.
